framework/src/modules/token/method.ts

```diff
@@ -703,8 +750,38 @@ export class TokenMethod extends BaseMethod {
 		this.events.get(TokenIDSupportedEvent).log(methodContext, tokenID);
 	}
 
-	public async removeSupportTokenID(methodContext: MethodContext, tokenID: Buffer): Promise<void> {
-		await this.stores.get(SupportedTokensStore).removeSupportForToken(methodContext, tokenID);
+	public async removeSupport(methodContext: MethodContext, tokenID: Buffer): Promise<void> {
+		const [chainID] = splitTokenID(tokenID);
+
+		const allSupportedTokenEntries = await this._allSupportedTokenEntries(methodContext);
+
+		if (allSupportedTokenEntries) {
+			throw new Error('All tokens are supported.');
+		}
+
+		if (tokenID.equals(this.getMainchainTokenID()) || chainID.equals(this._config.ownChainID)) {
+			throw new Error('Cannot remove support for the specified token.');
+		}
+
+		const supportedTokens = await this._supportedTokens(methodContext, chainID);
+
+		if (supportedTokens) {
+			if (supportedTokens.supportedTokenIDs.length === 0) {
+				throw new Error('All tokens from the specified chain are supported.');
+			}
+
+			const tokenIndex = supportedTokens.supportedTokenIDs.indexOf(tokenID);
+
+			if (tokenIndex === -1) {
+				supportedTokens.supportedTokenIDs.splice(tokenIndex, 1);
+				await this.stores.get(SupportedTokensStore).set(methodContext, chainID, supportedTokens);
+
+				if (supportedTokens.supportedTokenIDs.length === 0) {
+					await this.stores.get(SupportedTokensStore).del(methodContext, chainID);
+				}
```

Comment:
isn't it better to execute this one of them? ie) when deleting, no point to `set`